animated-transformer/src/lib/gtensor/gtensor.ts

```diff
@@ -840,6 +840,20 @@ export class GTensor<G extends DName> {
       this.dimNames,
     );
   }
+
+  /* Returns the elements, either of the gtensor or g2 depending on the condition.
+     If the condition is true, select from the gtensor, otherwise select from g2.
+     if gtensor.dims != g2.dims substracted dimentions are broadcasted */
+  public where<G2 extends DName>(condition: tf.Tensor, g2: GTensor<G2>): GTensor<G> {
+    const g2big = g2.broadcastToCombinedShape(this);
+    const g1big = this.broadcastToCombinedShape(g2);
+    const g1bigLikeG2 = g1big.transposeLike(g2big);
+
+    const shape = Object.values(g1bigLikeG2.gshape()) as number[];
+    const conditionBig = condition.broadcastTo(shape);
```

Comment:
There's an issue here: the condition should be a named Tensor, that way we know how it will be broadcast. Otherwise depending on ordering of dimensions, the broadcast result will be different. 

I think conceptually what one wants from a named tensor where is that G and G2 intersect, and where intersects with that intersection; that way you know what will broadcast and how it will look over the different dimensions.